src/hermit/mod.rs

```diff
@@ -28,6 +28,547 @@ pub type wctype_t = i64;
+pub const SOCK_STREAM: i32 = 1;
+pub const SOCK_NONBLOCK: i32 = 0o4000;
+pub const SOCK_CLOEXEC: i32 = 0o40000;
+pub const SOL_SOCKET: i32 = 4095;
+pub const SO_REUSEADDR: i32 = 0x0004;
+pub const SO_KEEPALIVE: i32 = 0x0008;
+pub const SO_BROADCAST: i32 = 0x0020;
+pub const SO_LINGER: i32 = 0x0080;
+pub const SO_SNDBUF: i32 = 0x1001;
+pub const SO_RCVBUF: i32 = 0x1002;
+pub const SO_SNDTIMEO: i32 = 0x1005;
+pub const SO_RCVTIMEO: i32 = 0x1006;
+pub const SO_ERROR: i32 = 0x1007;
+pub const TCP_NODELAY: i32 = 1;
+pub const MSG_PEEK: i32 = 1;
+pub const FIONBIO: i32 = 0x8008667eu32 as i32;
+pub const EAI_NONAME: i32 = -2200;
+pub const EAI_SERVICE: i32 = -2201;
+pub const EAI_FAIL: i32 = -2202;
+pub const EAI_MEMORY: i32 = -2203;
+pub const EAI_FAMILY: i32 = -2204;
+pub const POLLIN: i16 = 0x1;
+pub const POLLPRI: i16 = 0x2;
+pub const POLLOUT: i16 = 0x4;
+pub const POLLERR: i16 = 0x8;
+pub const POLLHUP: i16 = 0x10;
+pub const POLLNVAL: i16 = 0x20;
+pub const POLLRDNORM: i16 = 0x040;
+pub const POLLRDBAND: i16 = 0x080;
+pub const POLLWRNORM: i16 = 0x0100;
+pub const POLLWRBAND: i16 = 0x0200;
+pub const POLLRDHUP: i16 = 0x2000;
+pub const EFD_SEMAPHORE: i16 = 0o1;
+pub const EFD_NONBLOCK: i16 = 0o4000;
+pub const EFD_CLOEXEC: i16 = 0o40000;
+
+// sysmbols, which are part of the library operating system
+extern "C" {
+    /// Get the last error number from the thread local storage
+    #[link_name = "sys_get_errno"]
+    pub fn get_errno() -> i32;
+
+    /// If the value at address matches the expected value, park the current thread until it is either
+    /// woken up with [`futex_wake`] (returns 0) or an optional timeout elapses (returns -ETIMEDOUT).
+    ///
+    /// Setting `timeout` to null means the function will only return if [`futex_wake`] is called.
+    /// Otherwise, `timeout` is interpreted as an absolute time measured with [`CLOCK_MONOTONIC`].
+    /// If [`FUTEX_RELATIVE_TIMEOUT`] is set in `flags` the timeout is understood to be relative
+    /// to the current time.
+    ///
+    /// Returns -EINVAL if `address` is null, the timeout is negative or `flags` contains unknown values.
+    #[link_name = "sys_futex_wait"]
+    pub fn futex_wait(
+        address: *mut u32,
+        expected: u32,
+        timeout: *const timespec,
+        flags: u32,
+    ) -> i32;
+
+    /// Wake `count` threads waiting on the futex at `address`. Returns the number of threads
+    /// woken up (saturates to `i32::MAX`). If `count` is `i32::MAX`, wake up all matching
+    /// waiting threads. If `count` is negative or `address` is null, returns -EINVAL.
+    #[link_name = "sys_futex_wake"]
+    pub fn futex_wake(address: *mut u32, count: i32) -> i32;
+
+    /// sem_init() initializes the unnamed semaphore at the address
+    /// pointed to by `sem`.  The `value` argument specifies the
+    /// initial value for the semaphore.
+    #[link_name = "sys_sem_init"]
+    pub fn sem_init(sem: *mut *const c_void, value: u32) -> i32;
```

Comment:
```suggestion
    pub fn sem_init(sem: *mut sem_t, value: u32) -> i32;
```

Do we also want `pshared: i32` as argument?